src/main/java/tc/oc/pgm/listeners/ServerPingDataListener.java

```diff
@@ -0,0 +1,126 @@
+package tc.oc.pgm.listeners;
+
+import static com.google.common.base.Preconditions.*;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import java.util.UUID;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.server.ServerListPingEvent;
+import tc.oc.pgm.Config;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchManager;
+import tc.oc.pgm.api.match.event.MatchLoadEvent;
+import tc.oc.pgm.map.Contributor;
+import tc.oc.pgm.map.MapInfo;
+import tc.oc.pgm.map.PGMMap;
+import tc.oc.pgm.rotation.PGMMapOrder;
+
+public class ServerPingDataListener implements Listener {
+
+  private final MatchManager matchManager;
+  private boolean ready;
+
+  public ServerPingDataListener(MatchManager matchManager) {
+    this.matchManager = checkNotNull(matchManager);
+  }
+
+  @EventHandler
+  public void onMatchLoad(MatchLoadEvent event) {
+    ready = true;
+  }
+
+  @EventHandler
+  public void onServerListPing(ServerListPingEvent event) {
+    if (!Config.ServerPingData.enabled() || !ready) return;
+
+    JsonObject root = event.getOrCreateExtra(PGM.get());
+    for (Match match : this.matchManager.getMatches()) {
+      JsonObject matchObject = new JsonObject();
+      this.serializeMatch(match, matchObject);
+      root.add(match.getId(), matchObject);
+    }
```

Comment:
I think we should be caching this. We want server pings to be super fast.

Use a `Cache`

Reply:
done